azext_iot/_params.py

```diff
@@ -297,7 +297,7 @@ def load_arguments(self, _):
         )
         context.argument(
```

Comment:
assuming this param only applies to `az iot hub state`, I would move it to be under a with statement for "az iot hub state" rather than "iot hub"